pkg/services/publicdashboards/api/api.go

```diff
@@ -239,3 +250,15 @@ func toJsonStreamingResponse(features *featuremgmt.FeatureManager, qdr *backend.
 
 	return response.JSONStreaming(statusCode, qdr)
 }
+
+func validateRequest(req *PublicDashboardQueryDTO) error {
```

Comment:
We already have a file for putting request validation at `pkg/services/publicdashboards/validation/validator.go`. Could we put this in there?

Reply:
In principle I agree with this, the one caveat is that we're making additional queries even if the request is not valid, however, I think the trade off of having clean separation of logic is worthwhile. I just want to acknowledge we're making that tradeoff here.

Reply:
I like this idea, I can't see that caveat. What additional query we will be doing if we move this function?